src/SprykerEco/Zed/ProductManagementAi/Business/Generator/ImageAltTextGenerator.php

```
@@ -7,56 +7,73 @@
         ProductManagementAiConfig $productManagementAiConfig
     ) {
-        $this->openAiClient = $openAiClient;
-        $this->promptBuilder = $promptBuilder;
+        $this->aiFoundationClient = $aiFoundationClient;
         $this->productManagementAiConfig = $productManagementAiConfig;
     }
 
     /**
      * @param string $imageUrl
      * @param string $targetLocale
      *
-     * @return \Generated\Shared\Transfer\OpenAiChatResponseTransfer
+     * @return \Generated\Shared\Transfer\PromptResponseTransfer
      */
-    public function generateImageAltText(string $imageUrl, string $targetLocale): OpenAiChatResponseTransfer
+    public function generateImageAltText(string $imageUrl, string $targetLocale): PromptResponseTransfer
     {
-        $openAiChatRequestTransfer = (new OpenAiChatRequestTransfer())->setPromptData(
-            $this->promptBuilder->buildImageAltTextPrompt($imageUrl, $targetLocale),
-        )->setModel($this->productManagementAiConfig->getOpenAiGpt4oMiniModel());
+        $promptRequestTransfer = (new PromptRequestTransfer())
+            ->setPromptMessage(
+                (new PromptMessageTransfer())
+                    ->setContent($this->productManagementAiConfig->getImageAltTextPrompt($targetLocale))
+                    ->addAttachment((new AttachmentTransfer())
+                        ->setType(AiFoundationConstants::ATTACHMENT_TYPE_IMAGE)
+                        ->setContentType(AiFoundationConstants::ATTACHMENT_CONTENT_TYPE_URL)
+                        ->setType(AiFoundationConstants::ATTACHMENT_TYPE_IMAGE)
+                        ->setContent($imageUrl)),
+            );
+
+        try {
+            $promptResponseTransfer = $this->aiFoundationClient->prompt($promptRequestTransfer);
+        } catch (Exception $exception) {
+            $this->getLogger()->critical($exception->getMessage(), $exception->getTrace());
+
+            return (new PromptResponseTransfer())
```

this is not a proper handling, we should be transparent here set and then handle error explaining that something went wrong with the request to the model (optional: or even better explain what is wrong). Hiding error and setting empty sting might confuse way more then an exception